### PR TITLE
:sparkles: Implements Ratelimiter to prevent 429s from `user` scope

### PR DESCRIPTION
**Pull request by Lunarmagpie**
### Changes

-   `adds`: RATE LIMITING 
User scope is prevented by the ratelimiter. The shared scope is impossible to prevent according to some dude from the discord API server so I can ignore that one. I did not implement a way to prevent global rate limits in this PR.
All logging is under Info scope. Someone say something if you think it should be under a different one.

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

File: pincer/core/ratelimiter.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+from asyncio import sleep
+from dataclasses import dataclass
+import logging
+from time import time
+from typing import TYPE_CHECKING, Tuple
+
+if TYPE_CHECKING:
+    from typing import Dict
+    from .http import HttpCallable
```
Comment (trag1c) [inline pincer/core/ratelimiter.py:14]:
```suggestion
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import Dict, Tuple
    from .http import HttpCallable
```

**Review by trag1c — APPROVED** [review]

File: pincer/core/http.py [hunk 2]
```diff
@@ -63,7 +65,7 @@ def __init__(self, token: str, *, version: int = None, ttl: int = 5):
 
         version:
             The discord API version.
-            See `<https://discord.com/developers/docs/reference#api-versioning>`_.
```
Comment (RPSMain) [inline pincer/core/http.py:66]: Wouldn’t this break the link?

File: pincer/core/ratelimiter.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
+# Copyright Pincer 2021-Present
```
Comment (RPSMain) [inline pincer/core/ratelimiter.py:1]: Imo it would be better in like http it doesn’t really need its own file but idk

**Review by RPSMain — CHANGES_REQUESTED** [review]

File: pincer/core/ratelimiter.py (hunk 3 above)
Comment (Lunarmagpie) [reply pincer/core/ratelimiter.py:1]: No way. Http is already 500 lines which is close to really hard to work with imo.

File: pincer/core/http.py (hunk 2 above)
Comment (Lunarmagpie) [reply pincer/core/http.py:66]: Gotta check that. I forgot I had to.

Comment (Lunarmagpie) [reply pincer/core/http.py:66]: fixed that. Looks like it wasn't showing in the docs at all before :sweat_smile:

**Review by Endercheif — APPROVED** [review]

**Pull request merged**